ra2ce/analyses/analysis_config_wrapper.py

```diff
@@ -0,0 +1,185 @@
+"""
+                    GNU GENERAL PUBLIC LICENSE
+                      Version 3, 29 June 2007
+
+    Risk Assessment and Adaptation for Critical Infrastructure (RA2CE).
+    Copyright (C) 2023 Stichting Deltares
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+from typing import Optional
+
+from ra2ce.analyses.analysis_config_data.analysis_config_data import AnalysisConfigData
+from ra2ce.analyses.analysis_config_data.analysis_config_data_validator import (
+    AnalysisConfigDataValidator,
+)
+from ra2ce.common.configuration.config_wrapper_protocol import ConfigWrapperProtocol
+from ra2ce.graph.network_config_data.network_config_data_reader import (
+    NetworkConfigDataReader,
+)
+from ra2ce.graph.network_config_wrapper import NetworkConfigWrapper
+
+
+class AnalysisConfigWrapper(ConfigWrapperProtocol):
+    ini_file: Path
+    config_data: AnalysisConfigData
+    graphs: Optional[dict]
+
+    def __init__(self) -> None:
+        self.ini_file = None
+        self.config_data = AnalysisConfigData()
+        self.graphs = None
+
+    @staticmethod
+    def get_network_root_dir(filepath: Path) -> Path:
+        return filepath.parent.parent
+
+    @staticmethod
+    def get_data_output(ini_file: Path) -> Optional[Path]:
+        _root_path = AnalysisConfigWrapper.get_network_root_dir(ini_file)
+        _project_name = ini_file.parent.name
+        return _root_path.joinpath(_project_name, "output")
+
+    @property
+    def root_dir(self) -> Path:
+        return self.get_network_root_dir(self.ini_file)
```

Comment:
Would be ideal if we could move this logic into the `AnalysisConfigData` and leave the wrapper as just a 'controller' of the configuration `initialize_output_dirs` and so  on.

Reply:
Done